packages/cli-repl/src/cli-repl.ts

```diff
@@ -98,11 +177,29 @@ class CliRepl {
       // error is thrown here for atlas and DataLake connections.
       // don't actually throw, as this is only used to log out non-genuine
       // mongodb connections
-      this.bus.emit('mongodb:error', e)
+      this.bus.emit('mongosh:error', e)
       return null;
     }
   }
 
+  /**
+   * run buildInfo() command to get mongodbversion + other build info needed for
+   * logging
+   */
+  async getBuild(): Promise<any> {
+    // this could potentially error, so wrap it in a try/catch and format the
+    // returned error
+    try {
```

Comment:
Putting the `.catch` on `setupRepl` we can spare this try / catch, so the execution flow is interrupted where an error occur, otherwise it would continue.